### PR TITLE
Add DAO for user group and project member

### PR DESCRIPTION
**Pull request by stonezdj**
Add DAO for user group and project member

### CODE REVIEW

File: src/common/dao/group/usergroup.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	"github.com/vmware/harbor/src/common/dao"
+	"github.com/vmware/harbor/src/common/models"
+	"github.com/vmware/harbor/src/common/utils/log"
+)
+
+// AddUserGroup - Add User Group
+func AddUserGroup(userGroup models.UserGroup) (int, error) {
+	o := dao.GetOrmer()
+	query := models.UserGroup{GroupType: userGroup.GroupType, LdapGroupDN: userGroup.LdapGroupDN}
+	result, err := QueryUserGroup(query)
```
Comment (ywk253100) [inline src/common/dao/group/usergroup.go]: Maybe we should move the query logic to the API layer as we need to return 409?

Comment (stonezdj) [reply src/common/dao/group/usergroup.go]: Updated

File: src/common/dao/project/projectmember.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		return 0, fmt.Errorf("Invalid project_id, member: %+v", member)
+	}
+
+	sql := "insert into project_member (project_id, entity_id , role, entity_type) values (?, ?, ?, ?)"
+	_, err := o.Raw(sql, member.ProjectID, member.EntityID, member.Role, member.EntityType).Exec()
+	if err != nil {
+		return 0, err
+	}
+	var pmid int
+	querySQL := "select id from project_member where project_id = ? and entity_id = ? and entity_type = ? limit 1"
```
Comment (ywk253100) [inline src/common/dao/project/projectmember.go]: We can use sql.Result returned by Exec() to get the ID. Or use the beego ORM directly.

File: src/common/dao/project/projectmember.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	if err != nil {
+		return 0, err
+	}
+	return pmid, err
+}
+
+// UpdateProjectMemberRole updates the record in table project_member, only role can be changed
+func UpdateProjectMemberRole(pmID int, role int) error {
+	if role <= 0 || role >= 3 {
+		return fmt.Errorf("Failed to update project member role < 0, role:%v", role)
```
Comment (ywk253100) [inline src/common/dao/project/projectmember.go]: The error message should also contain role >= 3?

File: src/common/dao/project/projectmember.go (hunk 2 above)
Comment (stonezdj) [reply src/common/dao/project/projectmember.go]: Updated

File: src/common/dao/project/projectmember.go (hunk 3 above)
Comment (stonezdj) [reply src/common/dao/project/projectmember.go]: Updated

**Pull request merged**